src/sentry/tasks/store.py

```diff
@@ -486,6 +487,7 @@ def process_event_from_reprocessing(
 
 
 def _do_save_event(
+    consumer_type: str | None = None,
```

Comment:
Adding parameters to the front of the method signature could break existing call sites. It is generally safer to add new parameters to the end of a method so that you can't break existing callers.

Reply:
my 2c - feel free to ignore. at the front is more logical to me:
- it's the position the type is generally in elsewhere
- it really shouldn't have a default of none, we probably want to make this mandatory unlike the other args being passed here, so it should be first
- we have type checking to avoid breaking existing call sites so i don't think it's a real risk

Reply:
I don't disagree that the this parameter is makes the most sense in the first position.

> we have type checking to avoid breaking existing call sites so i don't think it's a real risk

Yeah for this function type-checking could identify problems, but type checking can't help with celery tasks.